graphbolt/src/cuda/common.h

```diff
@@ -9,9 +9,63 @@
 #include <c10/cuda/CUDAException.h>
 #include <cuda_runtime.h>
 
+#include <memory>
+#include <unordered_map>
+
 namespace graphbolt {
 namespace cuda {
 
+/*
+How to use this class to get a nonblocking thrust execution policy that uses
+torch's memory pool and the current cuda stream:
+
+cuda::CUDAWorkspaceAllocator allocator;
+const auto stream = torch::cuda::getDefaultCUDAStream();
+const auto exec_policy = thrust::cuda::par_nosync(allocator).on(stream);
+
+Now, one can pass exec_policy to thrust functions
+
+To get an integer array of size 1000 whose lifetime is managed by unique_ptr,
+use: auto int_array = allocator.alloc_unique<int>(1000); int_array.get() gives
+the raw pointer.
+*/
+class CUDAWorkspaceAllocator {
+  using ptr_map_t = std::unordered_map<void*, torch::Tensor>;
```

Comment:
is torch::Dict applied to the scenario?

Reply:
This class doesn't interact with anything torch-related, so it is more reasonable to use std::unordered_map here. It is a private member and it won't be exposed to python either.

Reply:
I think we will find a better solution in a future PR and this whole managing torch tensor lifetime logic will go away. This is a good enough solution to get started so that my future PRs will depend on this class.

Reply:
I will open an issue to simplify this class and get rid of the `std::unordered_map`.